### PR TITLE
feat: guard text inputs for CustomHeaderField component

### LINKED ISSUE #4557: feature request: Guard text inputs for `CustomHeaderField` component
### What problem did you meet?

In Logto, there is an input component (`CustomHeaderField`) for editing HTTP headers.
This component does not impose any constraints on the input content and the content is only guarded by the backend APIs:

image

### Describe what you'd like Logto to have

We would like to adhere to the [RFC 9110](https://www.rfc-editor.org/rfc/rfc9110.html#section-5) and improve our input component to validate and check for invalid inputs. The error reporting should be displayed in the following format:

image

### PR DESCRIPTION
**Pull request by hussainshaikh12**
## Summary

**Improved Header Validation in CustomHeaderField Component**

Changes:

1. Added validation for header key and value based on RFC 9110 standard to ensure:
    - Header keys use printable ASCII characters (excluding the colon).
    - Header values exclude control characters.
2. ValidHeaderKey and isValidHeaderValue functions were added to perform the validation
3. Fixes #4557
These changes ensure that the input headers in the CustomHeaderField component are compliant with expected standards.

## Testing

image
Tested the required CustomHeaderField component on the WebhookDetails page and it is working as expected.

- [x] This PR is not applicable for the checklist

### CODE REVIEW

File: packages/phrases/src/locales/en/translation/admin-console/webhook-details.ts [hunk 1]
```diff
@@ -40,6 +40,10 @@ const webhook_details = {
     key_duplicated_error: 'Key cannot be repeated.',
     key_missing_error: 'Key is required.',
     value_missing_error: 'Value is required.',
+    /** UNTRANSLATED */
```
Comment (wangsijie) [inline packages/phrases/src/locales/en/translation/admin-console/webhook-details.ts:43]: This tag is unnecessary for english language

File: packages/console/src/pages/WebhookDetails/WebhookSettings/CustomHeaderField/index.tsx [hunk 2]
```diff
@@ -70,7 +78,12 @@ function CustomHeaderField() {
                 placeholder="Key"
                 error={Boolean(headerErrors?.[index]?.key)}
                 {...register(`headers.${index}.key`, {
-                  validate: (key) => keyValidator(key, index),
+                  validate: (key) => {
+                    if (!isValidHeaderKey(key)) {
+                      return t('webhook_details.settings.invalid_key_error');
+                    }
```
Comment (xiaoyijun) [inline packages/console/src/pages/WebhookDetails/WebhookSettings/CustomHeaderField/index.tsx:84]: It seems that this check can be placed into the `keyValidator`.

File: packages/console/src/pages/WebhookDetails/WebhookSettings/CustomHeaderField/index.tsx [hunk 3]
```diff
@@ -79,10 +92,14 @@ function CustomHeaderField() {
...
-                    getValues(`headers.${index}.key`)
+                  validate: (value) => {
+                    if (!isValidHeaderValue(value)) {
+                      return t('webhook_details.settings.invalid_value_error');
+                    }
+                    return getValues(`headers.${index}.key`)
                       ? Boolean(value) || t('webhook_details.settings.value_missing_error')
-                      : true,
+                      : true;
+                  },
```
Comment (xiaoyijun) [inline packages/console/src/pages/WebhookDetails/WebhookSettings/CustomHeaderField/index.tsx:102]: Maybe adding a `valueValidator` would be better.

**Review by xiaoyijun — APPROVED** [review]
LGTM

**Review by wangsijie — APPROVED** [review]

**Comment by xiaoyijun** [comment]
close #4557

**Comment by charIeszhao** [comment]
Thank you for your contribution!

**Pull request merged**